cmd/kubeadm/app/cmd/join_test.go

```diff
@@ -206,17 +213,43 @@ func TestNewJoinData(t *testing.T) {
 			flags: map[string]string{
 				options.CfgPath: configFilePath,
 			},
+			data: &joinData{
+				cfg: &kubeadmapi.JoinConfiguration{
+					TypeMeta: metav1.TypeMeta{Kind: "", APIVersion: ""},
+					NodeRegistration: kubeadmapi.NodeRegistrationOptions{
+						Name:                  "somename",
+						CRISocket:             "unix:///var/run/containerd/containerd.sock",
+						IgnorePreflightErrors: []string{"c", "d"},
+						ImagePullPolicy:       "IfNotPresent",
+						Taints:                []v1.Taint{{Key: "node-role.kubernetes.io/control-plane", Effect: "NoSchedule"}},
+					},
+					CACertPath: kubeadmapiv1.DefaultCACertPath,
+					Discovery: kubeadmapi.Discovery{
+						BootstrapToken: &kubeadmapi.BootstrapTokenDiscovery{
+							Token:                    "abcdef.0123456789abcdef",
+							APIServerEndpoint:        "1.2.3.4:6443",
+							UnsafeSkipCAVerification: true,
+						},
+						TLSBootstrapToken: "abcdef.0123456789abcdef",
+						Timeout:           &metav1.Duration{Duration: kubeadmapiv1.DefaultDiscoveryTimeout},
+					},
+					ControlPlane: &kubeadmapi.JoinControlPlane{
+						CertificateKey: "c39a18bae4a72e71b178661f437363da218a3efb83ddb03f1cd91d9ae1da41bd",
+					},
+				},
+				ignorePreflightErrors: sets.New("c", "d"),
+			},
```

Comment:
I would prefer to change it to something like below.
```
			name: "--node-name flags override config from file",
			flags: map[string]string{
				options.CfgPath:  configFilePath,
				options.NodeName: "joinName",
			},
			validate: func(t *testing.T, data *joinData) {
				// validate that node-name is overwritten
				if data.cfg.NodeRegistration.Name != "joinName" {
					t.Error("Invalid NodeRegistration.Name")
				}
				if diff := cmp.Diff(joinData1, data, cmp.AllowUnexported(joinData{}), cmpopts.IgnoreFields(joinData{}, "client", "initCfg", "cfg.ControlPlane.LocalAPIEndpoint")); diff != "" {
					t.Fatalf("newJoinData returned data (-want,+got):\n%s", diff)
				}
			},
```

`validate    func(*testing.T, *joinData)`  can do the https://github.com/kubernetes/kubernetes/pull/113574/files#diff-1966268cd39e61d4b28b4c316f2073c64cf2337ae62e0e75ebb44c3d9abcf9f5R343-R347 check.

Reply:
Done